js_modules/dagit/packages/core/src/partitions/__tests__/CreatePartitionDialog.test.tsx

```diff
@@ -63,17 +66,27 @@ describe('CreatePartitionDialog', () => {
   });
 
   it('Shows error state', async () => {
+    render(<Test mocks={[]} />);
+
+    const partitionInput = await screen.findByTestId('partition-input');
+    expect(screen.queryByTestId('warning-icon')).toBeNull();
+    userEvent.clear(partitionInput);
+    userEvent.type(partitionInput, 'invalidname\n\r\t');
+    await waitFor(() => {
+      expect(screen.getByTestId('warning-icon')).toBeVisible();
+    });
     await act(async () => {
-      render(<Test mocks={[]} />);
+      userEvent.clear(partitionInput);
+    });
+    await waitFor(() => {
+      expect(screen.queryByTestId('warning-icon')).toBeNull();
+    });
+    await act(async () => {
+      await userEvent.type(partitionInput, 'validname');
```

Comment:
do we need `act` around `userEvent.type` ? seems odd.

Reply:
Yeah oddly enough we do, these tests do some pretty wild stuff if we remove this await. You can actually get into a situation where it's still typing /from the previous test/ if you don't wait for it to complete here.